### PR TITLE
[CUMULUS-2841] forward port of :  add integration test (#2729)

### PR DESCRIPTION
**Pull request by Jkovarik**
Co-authored-by: Danielle Peters <danielle.o.peters@nasa.gov>**Summary:** Summary of changes

Addresses [CUMULUS-2841](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2841)

## Changes

** Forward port of integration test changes made in https://github.com/nasa/cumulus/pull/2729

- **CUMULUS-2841**
  - Add integration test to validate PDR node provider that requires password
    credentials succeeds on ingest

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: example/cumulus-tf/iam.tf [hunk 1]
```diff
@@ -13,6 +13,10 @@ data "aws_s3_bucket" "fake_s3_provider_alternate" {
   bucket = data.aws_cloudformation_export.fake_s3_provider_alternate.value
 }
 
+data "aws_s3_bucket" "fake_s3_pftp_host_configuration_bucket" {
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply example/cumulus-tf/iam.tf:16]: It did not, should it just be ftp?

**Review by flamingbear — CHANGES_REQUESTED** [review]
Two minor nits and request for a clean script.
Also, the seed files are kinda large, you could definitely just change the contents to be strings of 'sample met file' and 'sample hdf' file which would mean we move fewer bytes on set up, but that's also just a suggestion.

File: example/lambdas/ftpPopulateTestLambda/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "@cumulus/ftp-populate-test-lambda",
+  "version": "9.4.1",
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/lambdas/ftpPopulateTestLambda/package.json:3]: Actually the release script will not catch it unless it's up to date.   Will fix.   Functionally it doesn't matter, but for consistency we should.

File: example/cumulus-tf/iam.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/cumulus-tf/iam.tf:16]: c75d8ea572

File: example/lambdas/ftpPopulateTestLambda/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/lambdas/ftpPopulateTestLambda/package.json]: f5e2557203

File: example/lambdas/ftpPopulateTestLambda/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  "description": "FTP Population Utility Lambda",
+  "main": "index.js",
+  "private": true,
+  "engines": {
+    "node": ">=12.18.0"
+  },
+  "scripts": {
+    "build": "rm -rf dist && mkdir dist && cp -R granules ./dist/granules && ../../../node_modules/.bin/webpack",
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/lambdas/ftpPopulateTestLambda/package.json:11]: 5f64a44a31 should be sufficient.   Lerna handles node_modules.

File: example/lambdas/ftpPopulateTestLambda/index.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    targetReplacementString,
+  } = params;
+  let data;
+  if (replacements.length > 0) {
+    data = fs.readFileSync(file, 'utf8');
+    replacements.forEach((replacement) => {
+      data = replace(data, new RegExp(replacement.old, 'g'), replacement.new);
+    });
+  } else {
+    data = fs.readFileSync(file);
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/lambdas/ftpPopulateTestLambda/index.js]: On second thought, we're inconsistent in our usage in the codebase, so the consistency argument goes right out the window.    Updated in favor of explicit: 

2e7029f745

File: example/cumulus-tf/iam.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/cumulus-tf/iam.tf]: c75d8ea572

**Comment by Jkovarik** [comment]
@flamingbear  I believe I've addressed all of the feedback, take a look at let me know what you think.

File: example/lambdas/ftpPopulateTestLambda/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (flamingbear) [reply example/lambdas/ftpPopulateTestLambda/package.json]: The actual dependencies will work with lerna ?

[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/lambdas/ftpPopulateTestLambda/package.json]: No, that was just bad.     This should be better: a1bb543e94

It won't functionally matter as we're never actually using the npm modules, but still.

**Comment by Jkovarik** [comment]
@flamingbear  re: your original fixture comment, 169f750e1e

**Pull request merged**